ProcessLib/TwoPhaseFlowWithPrho/TwoPhaseFlowWithPrhoMaterialProperties.h

```diff
@@ -0,0 +1,278 @@
+                                const ProcessLib::SpatialPosition& pos,
+                                const double p, const double T,
+                                const double saturation) const;
+    double getDerivCapillaryPressure(const double t,
+                                     const ProcessLib::SpatialPosition& pos,
+                                     const double p, const double T,
+                                     const double saturation) const;
+    double getLiquidDensity(const double p, const double T) const;
+    double getGasDensity(const double p, const double T) const;
+    double getGasViscosity(const double p, const double T) const;
+    double getLiquidViscosity(const double p, const double T) const;
+    double getDerivGasDensity(double const p, double const T) const;
+    bool computeConstitutiveRelation(
+        double const t,
+        ProcessLib::SpatialPosition const& x_position,
+        double const PG,
+        double const X,
+        double& Sw,
+        double& X_m,
+        double& dsw_dpg,
+        double& dsw_dX,
+        double& dxm_dpg,
+        double& dxm_dX);
+
+protected:
+    /// A flag to indicate whether the reference member, _material_ids,
+    /// is not assigned.
+    const bool _has_material_ids;
+
+    std::unique_ptr<MaterialLib::Fluid::FluidProperty> _liquid_density;
+    std::unique_ptr<MaterialLib::Fluid::FluidProperty> _viscosity;
+    std::unique_ptr<MaterialLib::Fluid::FluidProperty> _gas_density;
+    std::unique_ptr<MaterialLib::Fluid::FluidProperty> _gas_viscosity;
+
+    /** Use two phase models for different material zones.
+    *  Material IDs must be given as mesh element properties.
+    */
+    MeshLib::PropertyVector<int> const& _material_ids;
+
+    int _current_material_id = 0;
+    std::vector<Eigen::MatrixXd> _intrinsic_permeability_models;
+    std::vector<std::unique_ptr<MaterialLib::PorousMedium::Porosity>>
+        _porosity_models;
+    std::vector<std::unique_ptr<MaterialLib::PorousMedium::Storage>>
+        _storage_models;
+    std::vector<
+        std::unique_ptr<MaterialLib::PorousMedium::CapillaryPressureSaturation>>
+        _capillary_pressure_models;
+    std::vector<
+        std::unique_ptr<MaterialLib::PorousMedium::RelativePermeability>>
+        _relative_permeability_models;
+
+private:
+    /// Calculates the 18x1 residual vector.
+    void calculateResidual(double const PG, double const X, double Sw,
+                           double X_m, ResidualVector& res);
+
+    /// Calculates the 18x18 Jacobian.
+    void calculateJacobian(double const t, ProcessLib::SpatialPosition const& x,
+                           double const PG, double const X, JacobianMatrix& Jac,
+                           double Sw, double X_m);
+
+    /**
+    * Complementary condition 1
+    * for calculating molar fraction of light component in the liquid phase
+    */
+    double Calc_equili_rho_m(double const PG, double const Sw,
```

Comment:
Function names must be in camel case. (With some very few exceptions, which is not the case here.)
Use full english words, if possible.